controllers/tokenupload_controller.go

```diff
@@ -76,9 +76,11 @@ func (r *TokenUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
+	err := r.reconcileRemoteSecret(ctx, uploadSecret)
```

Comment:
Could you add an explanation here for why this needs to happen before we delete the secret? I actually don't see the difference in behavior between the original and your changes to the order of operations here.

Reply:
Added it, please check whether it makes sense for you.